src/rdb.c

```diff
@@ -3531,16 +3532,35 @@ int rdbSaveToReplicasSockets(int req, rdbSaveInfo *rsi) {
 
     /* Collect the connections of the replicas we want to transfer
      * the RDB to, which are i WAIT_BGSAVE_START state. */
-    server.rdb_pipe_conns = zmalloc(sizeof(connection *) * listLength(server.replicas));
-    server.rdb_pipe_numconns = 0;
-    server.rdb_pipe_numconns_writing = 0;
+    int connsnum = 0;
+    connection **conns = zmalloc(sizeof(connection *) * listLength(server.replicas));
+    server.rdb_pipe_conns = NULL;
+    if (!direct) {
```

Comment:
maybe consolidate the terminology on "dual conn" or something similar. 
```suggestion
    if (!dual_conn) {
```